### PR TITLE
refactor(Item Lists Node): Delete duplicate code for sorting via code (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**

### CODE REVIEW

File: packages/nodes-base/nodes/ItemLists/V3/helpers/utils.ts [hunk 1]
```diff
@@ -57,3 +64,26 @@ export const prepareFieldsArray = (fields: string | string[], fieldName = 'Field
 		`The \'${fieldName}\' parameter must be a string of fields separated by commas or an array of strings.`,
 	);
 };
+
+const returnRegExp = /\breturn\b/g;
```
Comment (michael-radency) [inline packages/nodes-base/nodes/ItemLists/V3/helpers/utils.ts:68]: should this var be inside function?

File: packages/nodes-base/nodes/ItemLists/V3/helpers/utils.ts [hunk 2]
```diff
@@ -57,3 +64,26 @@ export const prepareFieldsArray = (fields: string | string[], fieldName = 'Field
...
+		);
+	}
+
+	const mode = this.getMode();
+	const vm = new NodeVM({
+		console: mode === 'manual' ? 'redirect' : 'inherit',
+		sandbox: { items },
+	});
+
+	return vm.run(`module.exports = items.sort((a, b) => { ${code} })`);
```
Comment (michael-radency) [inline packages/nodes-base/nodes/ItemLists/V3/helpers/utils.ts:88]: is it deliberately not async anymore?

File: packages/nodes-base/nodes/ItemLists/V3/helpers/utils.ts (hunk 1 above)
Comment (netroy) [reply packages/nodes-base/nodes/ItemLists/V3/helpers/utils.ts:68]: should be fine to leave it here. just an old habit of mine to cache constants outside functions. doesn't make much of a difference now a days.

File: packages/nodes-base/nodes/ItemLists/V3/helpers/utils.ts (hunk 2 above)
Comment (netroy) [reply packages/nodes-base/nodes/ItemLists/V3/helpers/utils.ts:88]: yeah. I don't see why anyone would be using promises in sorting code. but I should also not make this a breaking changes. will revert.

Comment (netroy) [reply packages/nodes-base/nodes/ItemLists/V3/helpers/utils.ts:88]: nevermind. I just remembered. in the old code `async` code doesn't work, as the `newItems.sort( (a,b) => { ${code} })` block creates it's own non-async arrow function, which prevents the user from using `await` in their code.

**Review by michael-radency — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.5.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.5.1)

**Pull request merged**